### PR TITLE
Return Options for new UnixMountEntry

### PR DESCRIPTION
**Pull request by sophie-h**

### CODE REVIEW

**Review by sdroege — APPROVED** [review]

File: gio/src/unix_mount_entry.rs [hunk 1]
```diff
@@ -9,30 +9,28 @@ use std::mem;
...
         unsafe {
             let mut time_read = mem::MaybeUninit::uninit();
-            let ret = from_glib_full(ffi::g_unix_mount_at(
+            let ret: Option<UnixMountEntry> = from_glib_none(ffi::g_unix_mount_at(
                 mount_path.as_ref().to_glib_none().0,
                 time_read.as_mut_ptr(),
             ));
-            let time_read = time_read.assume_init();
-            (ret, time_read)
+            ret.map(|x| (x, time_read.assume_init()))
```
Comment (sdroege) [inline gio/src/unix_mount_entry.rs:19]: Actually `time_read` is always valid, even if `NULL` is returned.

File: gio/src/unix_mount_entry.rs [hunk 2]
```diff
@@ -9,30 +9,28 @@ use std::mem;
...
         unsafe {
             let mut time_read = mem::MaybeUninit::uninit();
-            let ret = from_glib_full(ffi::g_unix_mount_for(
+            let ret: Option<UnixMountEntry> = from_glib_none(ffi::g_unix_mount_for(
                 file_path.as_ref().to_glib_none().0,
                 time_read.as_mut_ptr(),
             ));
-            let time_read = time_read.assume_init();
-            (ret, time_read)
+            ret.map(|x| (x, time_read.assume_init()))
```
Comment (sdroege) [inline gio/src/unix_mount_entry.rs:33]: Same here

**Review by sdroege — CHANGES_REQUESTED** [review]
Otherwise seems correct :)

File: gio/src/unix_mount_entry.rs (hunk 1 above)
Comment (sophie-h) [reply gio/src/unix_mount_entry.rs]: Okay, was to lazy to check the C code

File: gio/src/unix_mount_entry.rs [hunk 3]
```diff
@@ -9,10 +9,10 @@ use std::mem;
 
 impl UnixMountEntry {
     #[doc(alias = "g_unix_mount_at")]
-    pub fn new_at<P: AsRef<std::path::Path>>(mount_path: P) -> (UnixMountEntry, u64) {
+    pub fn new_at<P: AsRef<std::path::Path>>(mount_path: P) -> (Option<UnixMountEntry>, u64) {
         unsafe {
             let mut time_read = mem::MaybeUninit::uninit();
-            let ret = from_glib_full(ffi::g_unix_mount_at(
+            let ret = from_glib_none(ffi::g_unix_mount_at(
```
Comment (sdroege) [inline gio/src/unix_mount_entry.rs:15]: It's `transfer full` though, or is that annotation wrong too?

Comment (sophie-h) [reply gio/src/unix_mount_entry.rs]: Somehow I was thinking that none=Option. I will have a look if I did this mistake in other commits :|

**Review by sdroege — APPROVED** [review]

**Pull request merged**